hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/EnhancerImpl.java

```diff
@@ -143,8 +140,8 @@ public EnhancerImpl(final EnhancementContext enhancementContext, final ByteBuddy
 		}
```

Comment:
Should the Map<String,Resolution> in classFileLocator, be cleared after each enhance(String className, byte[] originalBytes) call to avoid keeping the 'originalBytes' in memory, for large deployments?  

Does the transformation of entity classes involves following (class) references to other entity classes?  If yes, then we probably cannot clear the Map<String,Resolution>, since they will be needed for the transformations that span multiple classes being referenced by ByteBuddy.

Reply:
@scottmarlow, good questions. I was wondering about your second point as well.

Reply:
IMO, if Hibernate is enhancing application entity class named 'A' that is associated with class 'B', regardless of which class is loaded/defined/enhanced first, when the other class is enhanced, the Map<String, Resolution> in ClassFileLocator, would be the (input) unenhanced class, instead of the enhanced (returned) class.  I think it is likely that we can clear the Map after each call to enhance(), since we wouldn't want to use the (unenhanced) entity class.  Also it is unlikely that enhancing more than the class that is passed to the enhance() method, at the same time, would work (instead we should only enhance each class separately.